src/entity.ts

```diff
@@ -130,7 +150,48 @@ export namespace entity {
        * @name Int#value
        * @type {string}
        */
-      this.value = value.toString();
+      if (typeCastOptions) {
+        if (
+          typeCastOptions.integerTypeCastFunction &&
+          typeof typeCastOptions.integerTypeCastFunction !== 'function'
+        ) {
+          throw new Error(
+            `integerTypeCastFunction is not a function or was not provided.`
```

Comment:
This is a small thing, but it doesn't look possible for this error to be thrown when `integerTypeCastFunction` was not provided. Maybe just this would work:

```js
this.typeCastFunction = typeCastOptions.integerTypeCastFunction;

if (typeof this.typeCastFunction !== 'function') {
  throw ...
}
```

Reply:
Done